modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/TableDescriptor.java

```diff
@@ -88,6 +86,18 @@ public int engineId() {
         return engineId;
     }
 
+    public TableColumnDescriptor[] columns() {
+        return columns;
+    }
+
+    public boolean isKeyColumn(String name) {
+        return Arrays.binarySearch(primaryKeyColumns, name) >= 0;
```

Comment:
primaryKeyColumns array is not sorted.

Reply:
@AMashenkov I think there similar methods for checking whether in your or Pavel's patches and I am going to use those  after you patches get merged.